### ISSUE TITLE
Units (km/mi, $ etc)

### ISSUE THREAD

**Issue by jspaine** [freeCodeCamp/pantry-for-good#159]
Admin should be able to set currency and distance units.

**Comment by ronaldblanco** [comment]
Hello;
I would like to help on this issues; could you give me more detailed information of what you need in the application?

Thanks.

**Comment by jspaine** [comment]
Hi, just another field on the settings model on the server and an input on the client settings page to set it. Then the route summary component (`client/modules/driver/components/driver-assignment/RouteSummary.js`) should convert to miles if necessary.

**Comment by ronaldblanco** [comment]
I will be working on this!

**Comment by ronaldblanco** [comment]
Does not exist (client/modules/driver/components/driver-assignment/RouteSummary.js)
Searching in  (client/modules/driver/*/*.js) I did not find a file that managed distance for driver nor model.

Could you tell me what file exactly are you talking about?
Is there a distance procedure some place or be must build from scratch?

Thanks.

**Comment by jspaine** [comment]
[Here](https://github.com/freeCodeCamp/pantry-for-good/blob/staging/client/modules/driver/components/driver-assignment/RouteSummary.js). Did you get the app running?

**Comment by ronaldblanco** [comment]
Yes

**Comment by ronaldblanco** [comment]
Sorry it looks like I did miss something creating my branch.
Thanks for your help.

**Comment by ronaldblanco** [comment]
Hello jspaine
Hey I am following the contribution Guidelines step by step (because I did something wrong last time) but in the "setup frecodecamp" section I do not find the sample.env file and when I try to run the app I have this errors:

>npm run only-once

npm ERR! Linux 4.9.17-c9
npm ERR! argv "/home/ubuntu/.nvm/versions/node/v6.11.2/bin/node" "/home/ubuntu/.nvm/versions/node/v6.11.2/bin/npm" "run" "only-once"
npm ERR! node v6.11.2
npm ERR! npm  v3.10.10
npm ERR! missing script: only-once
npm ERR! 
npm ERR! If you need help, you may report this error at:
npm ERR!     <https://github.com/npm/npm/issues>
npm ERR! Please include the following file with any support request:
npm ERR!     /home/ubuntu/workspace/npm-debug.log

> gulp

[22:09:03] Using gulpfile ~/workspace/gulpfile.js
[22:09:03] Task 'default' is not in your gulpfile
[22:09:03] Please check the documentation for proper gulpfile formatting

How can I address that??
I am following the guide step by step!!

**Comment by ronaldblanco** [comment]
Running > npm run dev

I have error:
Application started on port 8080
events.js:160
      throw er; // Unhandled 'error' event
      ^
Error: listen EADDRINUSE :::8080
    at Object.exports._errnoException (util.js:1020:11)
    at exports._exceptionWithHostPort (util.js:1043:20)
    at Server._listen2 (net.js:1258:14)
    at listen (net.js:1294:10)
    at Server.listen (net.js:1390:5)
    at NativeConnection.<anonymous> (/home/ubuntu/workspace/server/server.js:27:41)
    at NativeConnection.g (events.js:292:16)
    at emitNone (events.js:86:13)
    at NativeConnection.emit (events.js:185:7)
    at /home/ubuntu/workspace/node_modules/mongoose/lib/connection.js:753:13
    at connectCallback (/home/ubuntu/workspace/node_modules/mongoose/node_modules/mongodb/lib/mongo_client.js:467:5)
    at /home/ubuntu/workspace/node_modules/mongoose/node_modules/mongodb/lib/mongo_client.js:386:11
    at _combinedTickCallback (internal/process/next_tick.js:73:7)
    at process._tickCallback (internal/process/next_tick.js:104:9)
[12:23:11] [nodemon] app crashed - waiting for file changes before starting...

**Comment by jspaine** [comment]
It means the port is in use, something's already running on it. Try `pkill node` if you can't find where its running.

**Comment by ronaldblanco** [comment]
I am running the app in a cloud9 work-space, it is there some kind of particularity here??

**Comment by jspaine** [comment]
Oh! I don't know, you should still be able to kill the app from the command line and restart it.

**Comment by ronaldblanco** [comment]
I think the problems is that gulp is using the 8080 and that's the port the work-space assign to the application.

**Comment by jspaine** [comment]
I don't think c9 should be using 8080 itself but im not sure. Are you sure you didnt start it already? Do you see any running node processes if you type `pgrep -l node`?

**Comment by ronaldblanco** [comment]
![image](https://user-images.githubusercontent.com/7390703/28874747-24dee80c-7761-11e7-87ac-26d57eb731ee.png)

**Comment by ronaldblanco** [comment]
I am sure!

**Comment by jspaine** [comment]
Ah it's because there's an environment variable PORT=8080. The app uses it for the server port so the server and client both try to run on 8080.
You can run it with `PORT=3000 npm run dev`

**Comment by jspaine** [comment]
Hmm, it mostly works if you change webpack.config.dev.js
```
publicPath: 'http://localhost:8080/'
```
to
```
publicPath: `https://${process.env.C9_C9_HOSTNAME}/`
```

And add to the devServer section: 
```
devServer: {
  ...
  disableHostCheck: true,
}
```

And change client/store/middleware/api: `'http://localhost:8080/api/'` to `'https://appname-username.c9users.io/api/'`

But it's really slow to load and it's going to be difficult to fix the all the urls so it works locally, on heroku and on cloud9. Is it the only way you can do it?

**Comment by ronaldblanco** [comment]
Hey I made it run with the port 8081 and it start ok but I am getting "Cannot GET /" when I access with the explorer.

**Comment by ronaldblanco** [comment]
![image](https://user-images.githubusercontent.com/7390703/28887057-bd9a179c-7789-11e7-968b-257371a49592.png)

**Comment by ronaldblanco** [comment]
Any suggestion?

**Comment by jspaine** [comment]
The server has to run on port 3000 in development so do `PORT=3000 npm run dev'. And you have to do the things I listed above.

Running it locally isn't an option at all?

**Comment by ronaldblanco** [comment]
Hello I am trying locally today

**Comment by ronaldblanco** [comment]
On my windows

**Comment by ronaldblanco** [comment]
Running "gulp watch" and creating the .env file with PORT and NODE_ENV make the app to run (apparently fine) and I have the 2 ports 8080 and 3000 listening; but when I go to my localhost:3000 I have a "Cannot GET /".
![image](https://user-images.githubusercontent.com/7390703/28921886-d0548b32-7825-11e7-9d0c-4347794dedff.png)
![image](https://user-images.githubusercontent.com/7390703/28921899-e123f9a2-7825-11e7-93ba-9ec24108606e.png)

Any Idea? I am on windows 7 and I have node 8.

**Comment by jspaine** [comment]
Go to localhost:8080

**Comment by ronaldblanco** [comment]
Yes I have the app on the 8080.
Do I am accessing to the client then??

**Comment by jspaine** [comment]
Yeah the client runs on 8080 and the server on 3000. I'll clean this thread up since it seems to be working now.

**Comment by ronaldblanco** [comment]
Thanks!

**Comment by ronaldblanco** [comment]
Hello, I hope you are ok.
I did a pull request to fix this issue, so let me know if everything is fine.

Thanks

**Comment by ronaldblanco** [comment]
Hey I think I fixed and did another pull request!
